common-js/build.gradle

```diff
@@ -1,28 +1,84 @@
+ext {
+    kotlinx_html_version = "0.6.4" 
+}
+
+
 buildscript {
+    ext.kotlin_version = '1.2.60'
+    ext.serialization_version = '0.6.1'
+
     repositories {
         mavenCentral()
+        maven { url "https://plugins.gradle.org/m2/" }
+        jcenter()
+     maven { url "https://kotlin.bintray.com/kotlinx" }
     }
+
     dependencies {
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath 'com.moowork.gradle:gradle-node-plugin:1.2.0'
+        classpath "org.jetbrains.kotlinx:kotlinx-gradle-serialization-plugin:$serialization_version"
     }
 }
 
+apply plugin: 'com.moowork.node'
 apply plugin: 'kotlin-platform-js'
+apply plugin: 'kotlin2js'
+apply plugin: 'kotlinx-serialization'
+
+repositories {
+    jcenter()
+     maven { url "https://kotlin.bintray.com/kotlinx" }
+    mavenCentral()
+}
 
-compileKotlin2Js {
+[compileKotlin2Js, compileTestKotlin2Js]*.configure {
     kotlinOptions {
-        metaInfo = true
         sourceMap = true
+        sourceMapEmbedSources = "always"
         moduleKind = 'commonjs'
         main = "call"
+        metaInfo = true
     }
 }
 
-repositories {
-    mavenCentral()
-}
-
 dependencies {
+    compile "org.jetbrains.kotlinx:kotlinx-html-js:$kotlinx_html_version"
+    compile "org.jetbrains.kotlinx:kotlinx-serialization-runtime-js:$serialization_version"
     compile "org.jetbrains.kotlin:kotlin-stdlib-js:$kotlin_version"
+
+    testCompile "org.jetbrains.kotlinx:kotlinx-serialization-runtime-js:$serialization_version"
+    testCompile "org.jetbrains.kotlinx:kotlinx-html-js:$kotlinx_html_version"
```

Comment:
Are these `testCompile` lines necessary for dependencies that don't have `test` in the name? My impression was that you `compile` the regular dependency and then `testCompile` the `test-`dependency. I could be wrong though